lib/linguist/heuristics.rb

```diff
@@ -473,5 +473,14 @@ def call(data)
         Language["XML"]
       end
     end
+
+    disambiguate ".w" do |data|
+      if (data.include?("&ANALYZE-SUSPEND _UIB-CODE-BLOCK _CUSTOM _DEFINITIONS C-Win"))
+        Language["OpenEdge ABL"]
+      else
```

Comment:
We try to avoid this kind of fall back option unless the first language is sure to match the heuristic. I think the second heuristic here could be `^@(<|\w+\.)`.